3_train.smk

```diff
@@ -1,15 +1,52 @@
 
 configfile: "3_train/train_config.yaml"
 
-# Train model
+# Save config files to output folder
+rule save_config:
+    input:
+        process_config = "2_process/process_config.yaml",
+        train_config = "3_train/train_config.yaml"
+    output:
+        process_config = "3_train/out/{data_source}/{run_id}/process_config.yaml",
+        train_config = "3_train/out/{data_source}/{run_id}/train_config.yaml"
+    shell:
+        """
+        mkdir -p 3_train/out/{wildcards.data_source}/{wildcards.run_id}
+        cp {input.process_config} {output.process_config}
+        cp {input.train_config} {output.train_config}
+        """
+
+# Train a model and save weights and metadata
+# Example:
+# $ snakemake --cores all 3_train/out/mntoha/4/hidden_size-50_weights.pt
 rule train_model:
     input:
-        npz_filepath = "2_process/out/{data_source}/train.npz"
+        npz_filepath = "2_process/out/{data_source}/train.npz",
+        # Include configs as inputs in order to save them to the output folder
+        process_config = "3_train/out/{data_source}/{run_id}/process_config.yaml",
+        train_config = "3_train/out/{data_source}/{run_id}/train_config.yaml"
     output:
-        weights_filepath = "3_train/out/{data_source}/{run}/lstm.pt",
-        metadata_filepath = "3_train/out/{data_source}/{run}/metadata.npz"
+        weights_filepath = "3_train/out/{data_source}/{run_id}/{model_id}_weights.pt",
+        metadata_filepath = "3_train/out/{data_source}/{run_id}/{model_id}_metadata.npz"
     params:
-        config = config
+        config = config,
```

Comment:
So here, `config`, which refers to `"3_train/train_config.yaml"` is passed in `3_train/src/train.py` as an input to `save_metadata()` and therefore gets saved to the `metadata.npz` file in each run folder. That `.npz` file, in turn, is read in and saved through execution of the `summarize_trained_models` rule. However, the process config, `2_process/process_config.yaml`, while saved to the trained model output directory, is not ever written to a metadata `.npz` file and thus isn't included in the summary of trained models.  If that was your intent, then there's no issue -- I just wanted to  make note of that discrepancy since you are intentionally saving the process config to the model output directory alongside the train config.

Reply:
Disregard my previous comment - I remembered that the process config is actually [saved to the training data `.npz` files](https://github.com/USGS-R/lake-temperature-lstm-static/blob/main/2_process/src/training_data.py#L211-L215), so that   is captured and included in the metadata [here](https://github.com/AndyMcAliley/lake-temperature-lstm-static/blob/log-settings/3_train/src/train.py#L386-L392).

Reply:
Yes, that's right, both configs are combined and saved to the metadata `.npz`. I got worried there for a second...